qldpc/codes.py

```diff
@@ -675,6 +761,382 @@ def _minimize_weight_of_logical_op(
         self._logical_ops[pauli.index, logical_qubit_index] = logical_op
 
 
+# TODO: factor out conjugation and local Pauli transformations to a separate method
+# TODO: allow construction of codes with field > 2, but throw errors for unsupported functionality
+class CSSCode_gen(QuditCode):
+    """CSS qudit code, with separate X-type and Z-type parity checks.
+
+    In order for the X-type and Z-type parity checks to be "compatible", the X-type stabilizers must
+    commute with the Z-type stabilizers.  Mathematically, this requirement can be written as
+
+    H_x @ H_z.T == 0,
+
+    where H_x and H_z are, respectively, the parity check matrices of the classical codes that
+    define the X-type and Z-type stabilizers of the CSS code.
+
+    """
+
+    code_x: ClassicalCode  # X-type parity checks, measuring Z-type errors
+    code_z: ClassicalCode  # Z-type parity checks, measuring X-type errors
+    field_order: int  # The order of the field over which the CSS code is defined
+
+    # _logical_ops: npt.NDArray[np.int_] | None = None
+
+    def __init__(
+        self,
+        code_x: ClassicalCode | IntegerMatrix,
+        code_z: ClassicalCode | IntegerMatrix,
+        field: int = None,
+    ) -> None:
+        """Construct a CSS code from X-type and Z-type parity checks."""
+
+        if field is None:
+            if isinstance(code_x, ClassicalCode):
+                self.field_order = self.code_x._field_order
+            elif isinstance(code_z, ClassicalCode):
+                self.field_order = self.code_z._field_order
+            else:
+                self.field_order = DEFAULT_FIELD_ORDER
+
+        self.code_x = ClassicalCode(code_x, self.field_order)
+        self.code_z = ClassicalCode(code_z, self.field_order)
+        # if not self.code_x._field_order == self.code_z._field_order :
+        #    raise ValueError("The sub-codes provided for this CSSCode are over different fields")
```

Comment:
Why is this commented out?

Reply:
I think that check will always pass based on the above setup. An error, if it occurs, will be thrown while initializing the X or Z classical codes.